### PR TITLE
SALTO-5917 - Salesforce: remove coordinate fields from auto-layout flows

### PR DESCRIPTION
**Pull request by tomsellek**
When flows are in auto-layout mode, the coordinates of their components are meaningless, yet when they change they cause many diffs in deployments.
This PR removes these coordinate values and re-adds them as 0 for deployment (since they are required)

---
Testing: 
 - Fetch a flow in auto-layout mode. Observe the fields not being present in the NaCl files.
 - Deploy a flow with no coordinate fields. Observe a successful deployment. Observe no coordinate fields in the NaCl file post-deploy.

---
_Release Notes_: 
Salesforce: coordinate fields of flow components that belong to a flow with AUTO_LAYOUT_CANVAS mode are now removed.

---
_User Notifications_: 
Salesforce: coordinate fields of flow components that belong to a flow with AUTO_LAYOUT_CANVAS mode are now removed.

### CODE REVIEW

File: packages/salesforce-adapter/src/filters/flow_coordinates.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+]
+
+const isInCanvasAutoLayoutMode = (instance: InstanceElement): boolean => {
+  if (!instance.value.processMetadataValues) {
+    return false
+  }
+  const canvasMode = instance.value.processMetadataValues.find(
+    ({ name }: { name: string }) => name === 'CanvasMode',
+  )
+  return canvasMode?.value?.stringValue === 'AUTO_LAYOUT_CANVAS'
```
Comment (yelly) [inline packages/salesforce-adapter/src/filters/flow_coordinates.ts:62]: Isn't all of this just:
```
instance.value?.processMetadataValues?.find(
  ({ name }: { name: string }) => name === 'CanvasMode',
)?.value?.stringValue === 'AUTO_LAYOUT_CANVAS' ?? false
```
I guess it's a matter of preference but having this single boolean expression looks better to me.

File: packages/salesforce-adapter/src/filters/flow_coordinates.ts
Comment (yelly) [inline packages/salesforce-adapter/src/filters/flow_coordinates.ts:1]: Can you add some tests for this? I think just one matching case and one NOP case should be enough.

**Review by yelly — COMMENTED** [review]
Looking good, just need to add tests.

File: packages/salesforce-adapter/src/filters/flow_coordinates.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        instance.value = transformValuesSync({
+          values: instance.value,
+          type: instance.getTypeSync(),
+          transformFunc: removeCoordinatesFromAllSections,
+        })
+      })
+  },
+  preDeploy: async (changes) => {
+    changes
+      .filter(isInstanceOfTypeChangeSync(FLOW_METADATA_TYPE))
```
Comment (amitgoldarb) [inline packages/salesforce-adapter/src/filters/flow_coordinates.ts:121]: This is missing an `.filter(isInCanvasAutoLayoutMode)` section.

**Review by amitgoldarb — APPROVED** [review]
This LGTM!
Can we make sure we test this on flows manually before merging? Also, this needs to be behind an adapter.config flag with the default off. Sadly this needs to be behind a prod ticket as it will edit a lot of flows.
This probably also needs noise reduction before we turn it on.

**Comment by tomsellek** [comment]
> Can we make sure we test this on flows manually before merging?

Is the testing I described in the PR notes not enough?

File: packages/salesforce-adapter/src/filters/flow_coordinates.ts (hunk 2 above)
Comment (tomsellek) [reply packages/salesforce-adapter/src/filters/flow_coordinates.ts:121]: If it's not in auto-layout mode then the coordinates will be present and we won't add them. I figured I'd just save an iteration.

**Comment by tomsellek** [comment]
> ... this needs to be behind an adapter.config flag with the default off. Sadly this needs to be behind a prod ticket as it will edit a lot of flows.
> This probably also needs noise reduction before we turn it on.


I will merge this code with the feature default-disabled. I created https://salto-io.atlassian.net/browse/PROD-341 to track the production rollout.

**Pull request merged**